app/webpacker/controllers/products_controller.js

```diff
@@ -39,4 +47,45 @@ export default class extends ApplicationController {
       "loading"
     ));
   };
+
+  // +recordType+ can either be 'product' or 'variant'
+  #deleteByRecordType(recordType) {
+    const deletePath = document.querySelector(`#${recordType}-delete-modal #modal-confirm-button`).getAttribute('data-delete-path');
+    const elementToBeRemoved = this.#getElementToBeRemoved(deletePath, recordType);
+
+    const handleSlideOutAnimationEnd = async () => {
+      const csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+
+      try {
+        const response = await fetch(deletePath, {
+          method: 'DELETE',
+          headers: {
+            Accept: 'text/vnd.turbo-stream.html',
+            'X-CSRF-Token': csrfToken,
+          }
+        });
+        // need to render the turboStream message, that's why not throwing error here 
+        if(response.status === 500) elementToBeRemoved.classList.remove('slide-out');
+
+        const responseTurboStream = await response.text();
+        Turbo.renderStreamMessage(responseTurboStream);
+      } catch(error) {
+        console.error(error.message);
+        elementToBeRemoved.classList.remove('slide-out');
+      }
+      finally {
+        elementToBeRemoved.removeEventListener('animationend', handleSlideOutAnimationEnd);
```

Comment:
Can handleSlideOutAnimationEnd be a private function below? To add the event listener, it will need `bind`:
```suggestion
        elementToBeRemoved.removeEventListener('animationend', this.#handleSlideOutAnimationEnd.bind(this));
```

Reply:
I think that won't work, we have to add and remove the same function. bind will return the new function everytime.